crates/core/transaction/src/is_action.rs

```diff
@@ -367,9 +367,40 @@ impl IsAction for Swap {
                         .cloned(),
                 }
             }
-            None => SwapView::Opaque {
-                swap: self.to_owned(),
-            },
+            None => {
+                // If we can find a matching BSOD in the TxP, use it to compute the output notes
+                // for the swap.
+                let bsod = txp
+                    .batch_swap_output_data
+                    .iter()
+                    // This finds the first matching one; there should only be one
+                    // per trading pair per block and we trust the TxP provider not to lie about it.
```

Comment:
> we trust the TxP provider not to lie about it

That's not actually a security concern, right? i.e., if the TxP provider (such as a wallet extension) were to put inaccurate data into the TxP, that'd just mean that the transaction would render as a view incorrectly, right? Just making sure I understand the stakes.

Reply:
Yes. One use case of the TxP that we are vaguely aiming at but haven't really explored in detail yet is as a selective disclosure mechanism to third parties (rather than untrusted first-party agents).

In the current scenario, we're doing selective disclosure to a first-party agent (the frontend) who's not trusted to have key material.

In a future scenario, we'd like to use the same mechanism to do disclosure to third parties, who may not trust the disclosing party to be honest. Thus, instead of consuming a view directly, they'd receive a TxP from the user, and then use that to generate their own view. This is mostly already possible. In this context, they'd probably want to fill in the public "bonus data" themselves, or check it against the chain.